craft_providers/util/retry.py

```diff
@@ -31,28 +31,32 @@ def retry_until_timeout(
     retry_wait: float,
     func: Callable[[float], T],
     *,
-    error: Exception = TimeoutError(),
+    error: Exception | None = TimeoutError(),
 ) -> T:
     """Re-run a function until it either succeeds or it times out.
 
     :param timeout: The length of time (in seconds) before timeout.
     :param retry_wait: The length of time (in seconds) before retrying.
     :param func: The callable. May only take a timeout parameter.
         Must raise an exception on failure, may return anything on success.
-    :param error: Exception to raise on timeout
+    :param error: Exception to raise on timeout or None to pass the error unchanged.
     :returns: The result of the function
     :raises: the passed error from the last exception
     """
     deadline = time.monotonic() + timeout
+    print(f"deadline {deadline=}")
```

Comment:
These `print`s are temporary / will be removed before merging?

Reply:
Ope, yep!